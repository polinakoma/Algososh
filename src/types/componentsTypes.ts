import { ElementStates } from "./element-states";

// string
export interface ISymbolsStep {
  symbols: string[];
  index?: number;
  state?: ElementStates;
};

// sorting-page
export interface INumbersArray {
  num: number;
  state?: ElementStates;
};

// stack-page
export interface IStack<T> {
  push: (item: T) => void;
  pop: () => void;
  clear: () => void;
};

export interface IStackSymbols {
  symbol: string | undefined,
  state: ElementStates
};

// queue-page 
export interface IQueueSymbols {
  symbol: string | undefined,
  state: ElementStates,
  head?: string | null,
  tail?: string | null,
};

// list-page
export interface ILinkedList<T> {
  append: (value: T) => void;
  prepend: (value: T) => void;
  addByIndex: (value: T, index: number) => void;
  deleteByIndex: (index: number) => void;
  deleteHead: () => void;
  deleteTail: () => void;
  getHead: () => void;
};

export interface IListSymbols {
  symbol: string | undefined,
  state: ElementStates,
  head?: 'head' | null,
  tail?: 'tail' | null,
  smallCircle?: IListSmallSymbol
};

export interface IListSmallSymbol {
  symbol: string | undefined
};