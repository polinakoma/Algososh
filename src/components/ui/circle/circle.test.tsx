import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";
import { Circle } from "./circle";

describe('Circle', () => {
    it('is empty', () => {
        const circle = renderer
        .create(<Circle/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('has symbol', () => {
        const circle = renderer
        .create(<Circle letter="symbol"/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('has head', () => {
        const circle = renderer
        .create(<Circle head='head'/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('has react-element in head', () => {
        const circle = renderer
        .create(<Circle head={<Circle/>}/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('has tail', () => {
        const circle = renderer
        .create(<Circle tail='tail'/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('has react-element in tail', () => {
        const circle = renderer
        .create(<Circle tail={<Circle/>}/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('has index', () => {
        const circle = renderer
        .create(<Circle index={1}/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('has prop isSmall === true', () => {
        const circle = renderer
        .create(<Circle isSmall/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('is default', () => {
        const circle = renderer
        .create(<Circle state={ElementStates.Default}/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('is changing', () => {
        const circle = renderer
        .create(<Circle state={ElementStates.Changing}/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
    it('is modifiend', () => {
        const circle = renderer
        .create(<Circle state={ElementStates.Modified}/>)
        .toJSON();
        expect(circle).toMatchSnapshot();
    });
});