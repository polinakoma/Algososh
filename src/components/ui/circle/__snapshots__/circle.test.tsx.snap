// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle has head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle has index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle has prop isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle has react-element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle has react-element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle has symbol 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      symbol
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle has tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle is changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle is default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle is empty 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle is modifiend 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
